vendor/k8s.io/utils/trace/trace.go

```diff
@@ -88,32 +88,27 @@ func (t *Trace) logWithStepThreshold(stepThreshold time.Duration) {
 	endTime := time.Now()
 
 	totalTime := endTime.Sub(t.startTime)
-	buffer.WriteString(fmt.Sprintf("Trace[%d]: %q ", tracenum, t.name))
 	if len(t.fields) > 0 {
 		writeFields(&buffer, t.fields)
-		buffer.WriteString(" ")
 	}
-	buffer.WriteString(fmt.Sprintf("(started: %v) (total time: %v):\n", t.startTime, totalTime))
+	klog.InfoS("Trace", "tracenum", tracenum, "name", t.name, "keyvalues", buffer.String(), "starttime", t.startTime, "totaltime", totalTime)
 	lastStepTime := t.startTime
 	for _, step := range t.steps {
 		stepDuration := step.stepTime.Sub(lastStepTime)
 		if stepThreshold == 0 || stepDuration > stepThreshold || klog.V(4).Enabled() {
-			buffer.WriteString(fmt.Sprintf("Trace[%d]: [%v] [%v] ", tracenum, step.stepTime.Sub(t.startTime), stepDuration))
-			buffer.WriteString(step.msg)
+			buffer.Reset()
 			if len(step.fields) > 0 {
-				buffer.WriteString(" ")
 				writeFields(&buffer, step.fields)
```

Comment:
Please flatten fields and pass them into klog.InfoS call

```
func logFields(fields []Field) []interface {
  result := make([]interface, 0, 2*len(fields)
  for _, field := range fields { 
    result = append(result, f.Key, f.Value)  
  }
  return result
}

klog.InfoS("Trace", "tracenum", tracenum, "steptime", step.stepTime.Sub(t.startTime), "stepduration", stepDuration, "stepmsg", step.msg, logFields(step.fields)...)
```

Reply:
In this case, I see the following error.
```
have (string, string, int32, string, time.Duration, string, time.Duration, string, string, []interface {}...)
want (string, ...interface {})
```

Can we use slice of interface{} after multiple key and values?
or do other values (startTime, traceNum, etc) also need to be included in the same slice as fields, making it **a single slice**?

Reply:
We need to put whole log into single slice (forgot about this)
```
klog.InfoS("Trace step", logStepArgs(step)...)

func logStepArgs(traceNum  int, step Step, stepDuration time.Duration) []interface {
  result := make([]interface, 0, 2*len(step.fields) + 8
  result = append(result, "traceNum", tracenum, "stepTime", step.stepTime.Sub(t.startTime), "stepDuration", stepDuration, "stepMsg", step.msg)
  for _, field := range fields { 
    result = append(result, f.Key, f.Value)  
  }
  return result
}
```

Reply:
I see, thanks